modules/src/core/ics02_client/msgs/update_client.rs

```diff
@@ -17,14 +20,17 @@ pub const TYPE_URL: &str = "/ibc.core.client.v1.MsgUpdateClient";
 
 /// A type of message that triggers the update of an on-chain (IBC) client with new headers.
 #[derive(Clone, Debug, PartialEq)] // TODO: Add Eq bound when possible
-pub struct MsgUpdateAnyClient {
+pub struct MsgUpdateAnyClient<C: ClientTypes> {
```

Comment:
We should  create an `AnyClientTypes` that implements ClientTypes so messages don't have to be generic